### ISSUE TITLE
[COGNITION] Smooth scrolling SHOULD be stopped when reduced motion is preferred

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/vets-design-system-documentation#1604]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

[Transferring @Mottie 's ticket from vagov](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6701) to the design system repo

## Description

To improve accessibility for all users, we **SHOULD** respect the accessibility preference to reduce motion option when set.

When enabled, we need to stop all smooth scrolling animation.
When disabled, smooth scrolling _should_ be okay.

## Acceptance Criteria

- Smooth scrolling is disabled when the OS reduce motion accessibility option is set.

## Environment

* Operating System: All
* Browser: All
* Screenreading device: N/A
* Server destination: All

## Steps to Recreate

1. Log in and open _any_ form
1. On a page where user input is expected, press the "Continue" button.
1. The browser will show an error and scroll the page to place the error at the top of the page
1. Or pressing "Continue" on a valid page or "Back" will also smooth scroll to bring the `h1` to the top of the page.

## Possible Fixes (optional)

Since we're using the React scroll library to add scroll animation, adding a css media query to stop css animation will not suffice. Instead, a scroll duration of zero should be passed to the library  

Use a JavaScript method to test media query settings: 

```js
// matchMedia will return true (when OS reduce motion accessibility option is set)
const duration = window?.matchMedia("(prefers-reduced-motion: reduce)")?.matches
  ? 0
  : 500; // milliseconds
```

## WCAG or Vendor Guidance (optional)

* [WCAG animation from interactions success criterion](https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html)
* [MDN reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Testing_media_queries)

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details
Arguably, depending on where this exists, it can range from being a moderate issue to a severe one that may cause harm to Veterans. If this isn't something we can fix universally in code, is it something we could write in documentation or advise teams on?

**Comment by rsmithadhoc** [comment]
While I agree, smooth scrolling should be disabled when prefers reduced motion is on, this is not part of the design system. The design system no longer uses React and the only reference to React scroll I could find is in `vets-website`. I don't believe we can set something globally in the CSS library that would override those uses. It is on VFS teams to remove React scroll, or set it to adhere to the reduced motion setting, so I am closing this issue.
